TraderJoe-Bots/Large-Swap/src/agent.spec.ts

```diff
@@ -0,0 +1,516 @@
+  create2Pair(testTokens[2][0], testTokens[2][1], mockNetworkManager.factory, mockNetworkManager.pairInitCodeHash),
+  create2Pair(testTokens[3][0], testTokens[3][1], mockNetworkManager.factory, mockNetworkManager.pairInitCodeHash),
+  create2Pair(testTokens[4][0], testTokens[4][1], mockNetworkManager.factory, mockNetworkManager.pairInitCodeHash),
+];
+// Format: [sender, amount0In, amount1In, amount0Out, amount1Out, to]
+const testCases: [string, BigNumber, BigNumber, BigNumber, BigNumber, string][] = [
+  [
+    createAddress("0xab123"),
+    BigNumber.from("10"),
+    BigNumber.from("25"),
+    BigNumber.from("75"),
+    BigNumber.from("325"), // Greater than 150 threshold
+    createAddress("0xab321"),
+  ],
+  [
+    createAddress("0xac456"),
+    BigNumber.from("225"), // Greater than 200 threshold
+    BigNumber.from("25"),
+    BigNumber.from("75"),
+    BigNumber.from("15"),
+    createAddress("0xac789"),
+  ],
+  [
+    createAddress("0xad654"),
+    BigNumber.from("35"),
+    BigNumber.from("25"),
+    BigNumber.from("444"), // Greater than 300 threshold
+    BigNumber.from("15"),
+    createAddress("0xad987"),
+  ],
+  [
+    createAddress("0xad654"),
+    BigNumber.from("35"),
+    BigNumber.from("600"), // Greater than 350 threshold
+    BigNumber.from("45"),
+    BigNumber.from("15"),
+    createAddress("0xad987"),
+  ],
+  [
+    createAddress("0xae741"),
+    BigNumber.from("35"), // All lower than either
+    BigNumber.from("25"), // 400 or 450 thresholds
+    BigNumber.from("40"),
+    BigNumber.from("15"),
+    createAddress("0xae842"),
+  ],
+];
+// Format: [reserve0, reserve1, blockTimestampLast]
+const testReserves: [BigNumber, BigNumber, number][] = [
+  [BigNumber.from("1000"), BigNumber.from("1500"), 456],
+  [BigNumber.from("2000"), BigNumber.from("2500"), 891],
+  [BigNumber.from("3000"), BigNumber.from("3500"), 765],
+  [BigNumber.from("4000"), BigNumber.from("4500"), 852],
+  [BigNumber.from("5000"), BigNumber.from("5000"), 753],
+];
+const testBlocks: number[] = [444, 3230, 90059, 90210, 230608];
+
+describe("Large Swap test suite", () => {
+  let handleTransaction: HandleTransaction;
+
+  const createTryAggregateCall = (
+    multicall2Address: string,
+    requireSuccess: boolean,
+    calls: [string, string][],
+    returnData: [boolean, string][],
+    blockNumber: number
+  ) => {
+    mockProvider.addCallTo(multicall2Address, blockNumber, MULTICALL2_IFACE, "tryAggregate", {
+      inputs: [requireSuccess, calls],
+      outputs: [returnData],
+    });
+  };
+
+  beforeAll(() => {
+    handleTransaction = provideHandleTransaction(testThresholdPercentage);
+  });
+
+  beforeEach(async () => {
+    mockProvider.clear();
+    // @ts-expect-error
+    mockProvider.getNetwork = jest.fn().mockReturnValue({ chainId: mockNetworkManager.chainId });
+    const initialize = provideInitialize(mockNetworkManager, mockProvider as any);
+    await initialize();
```

Comment:
It's better to avoid calling `initialize` inside tests. As mentioned in the previous comment, this can be avoided If the `networkManager` and `multicall2` are passed as parameters.